pkg/distributors/contracts/PoolTokenManipulator.sol

```diff
@@ -30,13 +30,20 @@ contract PoolTokenManipulator {
     mapping(bytes32 => EnumerableSet.AddressSet) private _poolTokenSets;
     mapping(bytes32 => bool) private _poolTokenSetSaved;
 
+    function savePoolTokenSet(bytes32 poolId) public {
+        (IERC20[] memory poolTokens, , ) = vault.getPoolTokens(poolId);
+        if (_poolTokenSetSaved[poolId]) {
+            delete _poolTokenSets[poolId];
```

Comment:
Have you tested this? I'm not sure it does what you want, since `delete` leaves mappings untouched: https://docs.soliditylang.org/en/v0.7.4/types.html#delete (a set is composed of a mapping and an array)

See [this issue on the OZ repo](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2607) for more discussion on clearing sets, particularly [this comment](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2607#issuecomment-829340362).

Reply:
Great catch.  Fixed to loop through and remove the tokens, and added a test using a `MockPool` with mutable tokens.